python/pyarrow/tests/conftest.py

```diff
@@ -84,16 +86,15 @@ def pytest_configure(config):
 
 def pytest_addoption(parser):
     for group in groups:
-        parser.addoption('--{0}'.format(group), action='store_true',
-                         default=defaults[group],
-                         help=('Enable the {0} test group'.format(group)))
+        for flag in ['--{0}', '--enable-{0}']:
```

Comment:
`--enable-hypothesis` is required, because `--hypothesis` option would match multiple options set by the hypothesis pytest plugin.